src/hotspot/share/gc/shenandoah/c2/shenandoahBarrierSetC2.cpp

```diff
@@ -944,6 +940,7 @@ void ShenandoahBarrierSetC2::clone_at_expansion(PhaseMacroExpand* phase, ArrayCo
     Node* gc_state    = phase->transform_later(new LoadBNode(ctrl, mem, gc_state_addr, gc_state_adr_type, TypeInt::BYTE, MemNode::unordered));
     int flags = ShenandoahHeap::HAS_FORWARDED;
     if (ShenandoahIUBarrier) {
+      // TODO: Shouldn't this be OLD_MARKING too?
```

Comment:
Generational mode requires SATB barriers right now - IU mode only has the young generation.

Reply:
Ok, removed.